scripts/ilab-macos-installer.sh

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env bash
+
+# Simple usage guide:
+# This script is designed for macOS users, specifically optimized for Apple Silicon (M1, M2, M3) users.
+# It will install InstructLab in a Python virtual environment with support for Apple Metal (MPS) acceleration
+# for enhanced performance on M-series Macs.
+
+# If you're using an Intel-based Mac, this script will not work, as it is tailored for Apple Silicon (arm64) architecture.
+#
+# Download and run the script:
+#    curl -s https://raw.githubusercontent.com/instructlab/instructlab/refs/heads/main/scripts/ilab-macos-installer.sh | bash
+#
+# Specify INSTALL_DIR environment variable for installation directory
+#    curl -s https://raw.githubusercontent.com/instructlab/instructlab/refs/heads/main/scripts/ilab-macos-installer.sh | bash -s -- /install/path
+
+INSTALL_DIR=${1:-"$HOME/instructlab"}
+OS_TYPE=$(uname)
+OS_ARCH=$(uname -m)
+CPU_BRAND=$(sysctl -n machdep.cpu.brand_string)
+LINK_TARGET_DIR="/usr/local/bin"
```

Comment:
It gives me an error when I launch it, do I need to run it with `sudo`? 
```
Error: No write permission to /usr/local/bin.
Run the script with sudo or choose another directory.
```

Reply:
Thanks for you try. I think you may need to use `sudo` for the soft link.

Reply:
weird, only show two lines? Should be three
```
        echo -e "\nError: No write permission to $LINK_TARGET_DIR."
        echo "Run the script with sudo or choose another directory."
        echo -e "Alternatively, you can use sudo create the symlink manually for $INSTALL_DIR/venv/bin/ilab"
```

Reply:
Of course I can use `sudo`, as you say, but when I run the full unit test suite with `tox -e py3-unit` it also runs this test which fails for the same reason. Maybe we can remove it from the pytest suite? (easiest way is to rename it to something like `macos_install.py`)

Reply:
Thank you so much for feedback. But based on the previous discussion, the test is needed for the script.
I try to fix with the venv whole path(will not exit) to run the next steps if no permission, make sure it can finish to the end. I also tried simply use `sudo` with curl command, seems will totally change the dir permission, that's not good to the current user. 
can you try below? 
[2827](https://github.com/instructlab/instructlab/pull/2827)